### PR TITLE
upgrade corenlp to the latest revision 3.8.0

### PR DESCRIPTION
**Pull request by dispalt**
Just some small changes to method signatures and objects.

### CODE REVIEW

File: corenlp/src/main/scala/org/clulab/processors/fastnlp/FastNLPProcessor.scala [hunk 1]
```diff
@@ -61,8 +61,8 @@ class FastNLPProcessor(
       val gs = stanfordDepParser.predict(sa)
 
       // convert to Stanford's semantic graph representation
-      val basicDeps = SemanticGraphFactory.makeFromTree(gs, SemanticGraphFactory.Mode.BASIC, GrammaticalStructure.Extras.NONE, true, null)
-      val collapsedDeps = SemanticGraphFactory.makeFromTree(gs, SemanticGraphFactory.Mode.CCPROCESSED, GrammaticalStructure.Extras.NONE, true, null)
+      val basicDeps = SemanticGraphFactory.makeFromTree(gs, SemanticGraphFactory.Mode.BASIC, GrammaticalStructure.Extras.NONE, null)
+      val collapsedDeps = SemanticGraphFactory.makeFromTree(gs, SemanticGraphFactory.Mode.CCPROCESSED, GrammaticalStructure.Extras.NONE, null)
```
Comment (dispalt) [inline corenlp/src/main/scala/org/clulab/processors/fastnlp/FastNLPProcessor.scala]: The `threadSafe` parameter was removed.

File: corenlp/src/main/scala/org/clulab/processors/corenlp/CoreNLPProcessor.scala [hunk 2]
```diff
@@ -74,7 +74,7 @@ class CoreNLPProcessor(
       val stanfordTree = stanfordParse(sa)
 
       // store Stanford annotations; Stanford dependencies are created here!
-      ParserAnnotatorUtils.fillInParseAnnotations(false, true, gsf, sa, stanfordTree, GrammaticalStructure.Extras.NONE)
+      ParserAnnotatorUtils.fillInParseAnnotations(false, true, gsf, sa, util.Arrays.asList(stanfordTree), GrammaticalStructure.Extras.NONE)
```
Comment (dispalt) [inline corenlp/src/main/scala/org/clulab/processors/corenlp/CoreNLPProcessor.scala]: I am not sure about this one.

Comment (MihaiSurdeanu) [reply corenlp/src/main/scala/org/clulab/processors/corenlp/CoreNLPProcessor.scala]: Seems like the right change.

**Comment by MihaiSurdeanu** [comment]
Thanks Dan!
We are not quite ready for this change, because reach (our biomedical IE system) depends on Stanford dependencies, whereas this version of CoreNLP uses a different representation (universal dependencies). Let's keep this in a branch for a few weeks.

@hickst, @marcovzla, @myedibleenso: I would like to start a major version of processors (7.x) which uses the latest CoreNLP, and we do a complete move to universal dependencies. This means that Reach will have to stay with processors 6.x. Which further means that, if there are major changes to processors that would be needed in Reach (although I don't see any...), we would have to backport these changes to 6.x. Thoughts? Given the new WM project, I would like to do this asap.

**Comment by hickst** [comment]
UD is being worked on. Closing this PR for now.

**Comment by MihaiSurdeanu** [comment]
Yes, we hope to have the update to 3.8.0 up in a couple of days. It turns
out it was more complicated than expected due to several hidden changes in
CoreNLP.

**Pull request closed without merging**